OrbitGl/TimeGraph.cpp

```diff
@@ -833,35 +579,29 @@ void TimeGraph::SelectEvents(float a_WorldStart, float a_WorldEnd,
   }
 
   NeedsUpdate();
+
+  return selected_callstack_events;
 }
 
 //-----------------------------------------------------------------------------
 void TimeGraph::Draw(bool a_Picking) {
-  // TODO: keep() used to maintain a cap on the number of timers we store in
-  // memory.
-  //       Now that each thread track has multiple blockchains as opposed to
-  //       only having a single global one, this a bit trickier.
-  if (/*m_TextBoxes.keep( GParams.m_MaxNumTimers ) ||*/ (
-          !a_Picking && m_NeedsUpdatePrimitives) ||
-      a_Picking) {
-    UpdatePrimitives(a_Picking);
+  if ((!a_Picking && m_NeedsUpdatePrimitives) || a_Picking) {
+    UpdatePrimitives();
   }
 
-  DrawThreadTracks(a_Picking);
+  DrawTracks(a_Picking);
   DrawBuffered(a_Picking);
 
   m_NeedsRedraw = false;
 }
 
 //-----------------------------------------------------------------------------
-void TimeGraph::DrawThreadTracks(bool a_Picking) {
+void TimeGraph::DrawTracks(bool a_Picking) {
   m_Layout.SetNumCores(GetNumCores());
-  const std::vector<ThreadID>& sortedThreadIds = m_Layout.GetSortedThreadIds();
-  for (ThreadID threadId : sortedThreadIds) {
-    std::shared_ptr<ThreadTrack> track = GetThreadTrack(threadId);
+  for (auto& track : sorted_tracks_) {
     if (track->GetName().empty()) {
       std::string threadName =
-          Capture::GTargetProcess->GetThreadNameFromTID(threadId);
+          Capture::GTargetProcess->GetThreadNameFromTID(track->GetID());
```

Comment:
Shouldn't this only be done for thread tracks? Or are sorted_tracks only thread tracks? (If yes, maybe clarify in the name.)

Reply:
Yes you are right.  Added a condition.